### ISSUE TITLE
'var' shouldn't infer nullable annotation for readonly variables

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#42702]
**Version Used**: 
```
Branch master (10 Mar 2020)
Latest commit 441c154 by msftbot[bot]:
Merge pull request #42303 from CyrusNajmabadi/useCompoundParens

Support parenthesized expressions in use-compound-operator.
```

**Steps to Reproduce**:

```cs
#nullable enable
public class C {
    public void M(string[] arr) {
        foreach (var x in arr)
        {
          void Local() => x.ToString();
        }
    }
}
```

**Expected Behavior**:
No warnings. There's not a single nullable type in the whole program.

**Actual Behavior**:
`warning CS8602: Dereference of a possibly null reference.` is reported for `x.ToString`

**Notes**
It looks like this was discussed at some point as mentioned there: https://github.com/dotnet/roslyn/issues/40903
@jcouv were local functions considered when this decision was made? I feel that incorrect diagnostics in local functions outweight any other argument mentioned there (regularity, diagnostic messages, etc) by far

**Comment by TessenR** [comment]
Seems like local functions are now using variables' state at the points of invocations, I missed this and tried an example with an unused local function. Guess it can be closed then.
